interfaces/repo.go

```diff
@@ -80,6 +80,112 @@ func (r *Repository) AddInterface(i Interface) error {
 	return nil
 }
 
+// QueryOptions describes options for QueryInterfaces.
+//
+// Names: return just this subset if non-empty.
+// Doc: return documentation.
+// Plugs: return information about plugs.
+// Slots: return information about slots.
+// Connected: only consider interfaces with at least one connection.
+type QueryOptions struct {
+	Names                        []string
+	Doc, Plugs, Slots, Connected bool
+}
+
+func (r *Repository) queryInterface(iface Interface, opts *QueryOptions) *InterfaceInfo {
+	// NOTE: QueryOptions.Connected is handled by QueryInterfaces
+	md := MetaDataOf(iface)
```

Comment:
Same as for the client, we need to either unify metadata and info, or give MetaData a better name. As a way to move forward and not have to refactor too much right away, I suggest calling MetaData as StaticInfo, which gives a more clear delta between the place of each of these in our world: StaticInfo is the static details provided at interface declaration time, and which we do have in Info as well.